### PR TITLE
fix: show upgrade banner only when needed

### LINKED ISSUE #2793: Bug: upgrade banner is displayed when not logged in
### Describe the bug

image

### Steps to reproduce

Visit a public workspace repository insight 

https://app.opensauced.pizza/workspaces/68b44538-57e3-467b-a374-81be64865fba/contributor-insights/759e58cd-2e89-4704-8def-93d911716b82/overview

The upgrade banner should only be displayed if logged in and an owner of the workspace

### Browsers

_No response_

### Additional context (Is this in dev or production?)

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [ ] I agree to follow this project's Contribution Docs

### LINKED ISSUE #2801: Bug: Upgrade to pro banner is displayed for pro account in Contributor Insights Page
### Describe the bug

Not sure if this is related to #2793 as the bug there is affecting users who are not logging in.

---

I have OpenSauced pro account and I found this bug while I logged in.

When I opened one of the page in the Contributor Insights, the banner of "Upgrade to a PRO Workspace" is displayed on the top. I notice, this bug is only affected the page that's been created by adding contributors manually through "Explore Contributor". 

**Additional info**:

- Another pages that have been created through "sync your GitHub team" and "importing GitHub following" are not affected.
- Repository Insights Pages are not affected. 

### Screenshots

Screenshot 2024-02-28 111423

Screenshot 2024-02-28 111439

### Steps to reproduce

1. Open the sidebar.
2. Click a page in "Contributor Insights" that's been created by adding contributors manually.
3. You will see the banner on top.

### Browsers

Chrome

### Additional context (Is this in dev or production?)

production

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [ ] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by zeucapua**
## Description

Fixes an issue where the "Upgrade to a PRO Workspace" banner is being shown to users not logged in, for insight pages that are made by someone else, and to those whose accounts are PRO but the workspace isn't. Also changes the banner to be full width.

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Closes #2793 
Closes #2801 

## Mobile & Desktop Screenshots/Recordings

From a non PRO account and their workspace insight
image

Same insight, but from another account
Screenshot 2024-02-29 at 10 10 47 AM

## Steps to QA

To see the banner, go to an insight where:
1. You are logged in and own the insight
2. You do NOT have a workspace OR account PRO subscription

To NOT see the banner, go to an insight where:
1. You are logged in and own the insight
2. You have either a PRO workspace for that insight OR a PRO account subscription
-- or --
1. You do not own the insight (whether or not you're logged in)

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

arrested-development-michael-bluth-gif

### CODE REVIEW

**Comment by zeucapua** [comment]
move banner to `WorkspaceLayout` and make full width, first commented here:
https://github.com/open-sauced/app/pull/2796#issuecomment-1969701083

**Comment by nickytonline** [comment]
> move banner to `WorkspaceLayout` and make full width, first commented here: [#2796 (comment)](https://github.com/open-sauced/app/pull/2796#issuecomment-1969701083)

If the banner can be displayed on any page, it's fine to put it there, but if not, I'd recommend keeping it to the individual pages. You could always use a React Portal for an element in the `<WorkspaceLayout />`. I only mention it as there is near future work to have the workspaces layout on every page.

**Comment by zeucapua** [comment]
> > move banner to `WorkspaceLayout` and make full width, first commented here: [#2796 (comment)](https://github.com/open-sauced/app/pull/2796#issuecomment-1969701083)
> 
> If the banner can be displayed on any page, it's fine to put it there, but if not, I'd recommend keeping it to the individual pages. You could always use a React Portal for an element in the `<WorkspaceLayout />`. I only mention it as there is near future work to have the workspaces layout on every page.

Figured any other pages for workspaces (in the future) that might need the banner can easily use the optional `bannerProps` to show/not show it. This was also just the first solution I got when making the banner full width. Would you like me to keep them on the individual pages instead?

**Comment by nickytonline** [comment]
> > > move banner to `WorkspaceLayout` and make full width, first commented here: [#2796 (comment)](https://github.com/open-sauced/app/pull/2796#issuecomment-1969701083)
> > 
> > 
> > If the banner can be displayed on any page, it's fine to put it there, but if not, I'd recommend keeping it to the individual pages. You could always use a React Portal for an element in the `<WorkspaceLayout />`. I only mention it as there is near future work to have the workspaces layout on every page.
> 
> Figured any other pages for workspaces (in the future) that might need the banner can easily use the optional `bannerProps` to show/not show it. This was also just the first solution I got when making the banner full width. Would you like me to keep them on the individual pages instead?

No, all good. If we need to adjust this, we can do it later. Thanks Zeu!

File: components/Workspaces/WorkspaceLayout.tsx [hunk 1]
```diff
@@ -60,8 +69,26 @@ export const WorkspaceLayout = ({ workspaceId, children }: WorkspaceLayoutProps)
           )}
         </ClientOnly>
       </div>
-      <div className="flex flex-col items-center grow pt-8 md:pt-14 lg:pt-20">
+      <div className="relative flex flex-col items-center grow pt-8 md:pt-14 lg:pt-20">
```
Comment (nickytonline) [inline components/Workspaces/WorkspaceLayout.tsx:72]: not what I mentioned about using React portals the other day, but I envisioned it more passing the component and use the `bannerProps` prop (probably rename it to `banner`) like a slot.

```jsx
<WorkspaceLayout workspaceId={workspaceId} banner={<p>Hello</p>}>
...
</WorkspaceLayout>
```

and in the WorkspaceLayout component.

have e.g. `banner` props is optional.

```jsx
export const WorkspaceLayout = ({ workspaceId, banner, children }: WorkspaceLayoutProps) => {
  return (
    <div>
      ///...
      <div className="relative flex flex-col items-center grow pt-8 md:pt-14 lg:pt-20">{banner}</div>
    </div>
  );
};
```

This way the `WorkspaceLayout is only responsible for layout and we can pop in whatever banner we want.

**Review by nickytonline — APPROVED** [review]
Thanks @zeucapua! 🚢

**Review by brandonroberts — APPROVED** [review]

**Pull request merged**